### PR TITLE
[core] add method to kill aws instance to simulate chaos

### PR DESCRIPTION
**Pull request by hongchaodeng**
## Why are these changes needed?

This is separated from https://github.com/ray-project/ray/pull/45364.
Have tested it for `chaos_test/test_chaos_basic.py`.

How to enable this feature is to be discussed? We can enable it for all chaos tests, maybe? Or we can add a flag to migrate tests one by one?

I will test it on more existing tests and discuss with other teams to decide later.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/test_utils.py [hunk 1]
```diff
@@ -1533,6 +1534,34 @@ def _kill_resource(self, node_id, node_to_kill_ip, node_to_kill_port):
...
+        instanceId=$(curl -H "X-aws-ec2-metadata-token: $TOKEN" http://169.254.169.254/latest/meta-data/instance-id/)
+        region=$(curl -H "X-aws-ec2-metadata-token: $TOKEN" http://169.254.169.254/latest/meta-data/placement/region)
+
+        aws ec2 terminate-instances --region $region --instance-ids $instanceId
+        """  # noqa: E501
+
+        ssh = paramiko.SSHClient()
+        ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+        # This is a feature on Anyscale platform that enables
```
[the remark this answers is not in the document]
Comment (hongchaodeng) [reply python/ray/_private/test_utils.py:1545]: That's something to add to the test. This is the utility method.

That's also why I separate this from https://github.com/ray-project/ray/pull/45364. I can't make the call and need to talk to other decision makers to follow up on config change.

File: python/ray/_private/test_utils.py [hunk 2]
```diff
@@ -1533,6 +1533,23 @@ def _kill_resource(self, node_id, node_to_kill_ip, node_to_kill_port):
             )
             self.killed.add(node_id)
 
+    def _kill_node(self, ip):
```
Comment (jjyao) [inline python/ray/_private/test_utils.py:1536]:
```suggestion
    def _terminate_ec2_instance(self, ip):
```

**Review by jjyao — APPROVED** [review]

File: python/ray/_private/test_utils.py [hunk 3]
```diff
@@ -1533,6 +1533,23 @@ def _kill_resource(self, node_id, node_to_kill_ip, node_to_kill_port):
...
+            'region=$(curl -H "X-aws-ec2-metadata-token: $TOKEN" http://169.254.169.254/latest/meta-data/placement/region);'  # noqa: E501
+            "aws ec2 terminate-instances --region $region --instance-ids $instanceId"  # noqa: E501
+        )
+        # This is a feature on Anyscale platform that enables
+        # easy ssh access to worker nodes.
+        ssh_command = f"ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -p 2222 ray@{ip} '{multi_line_command}'"  # noqa: E501
+
+        result = subprocess.run(ssh_command, shell=True, capture_output=True, text=True)
+        print(f"STDOUT:\n{result.stdout}\n")
+        print(f"STDERR:\n{result.stderr}\n")
```
Comment (rynewang) [inline python/ray/_private/test_utils.py:1551]: just for the sake of completeness, can we detect if the aws cli return 0 or not, and raise exceptions if it's not returning 0?

Comment (hongchaodeng) [reply python/ray/_private/test_utils.py:1551]: Done.
 It will returned CalledProcessError if failed.

**Review by rynewang — APPROVED** [review]

**Pull request merged**